rs-react/src/App.tsx

```diff
@@ -70,34 +47,27 @@ export function App() {
     setSearchParams({ page: String(newPage) });
   };
 
-  if (hasError) {
-    throw new Error('The button was clicked');
-  }
-
   const handleCardClick = (character: Character) => {
     setSelectedCharacter(character);
-  };
+  }
 
   const { gender, species, status, origin, created } = selectedCharacter ?? {};
+
+
   return (
     <>
       <Header onSearchSubmit={handleSearchSubmit} />
       <div id='main'>
         <div id='side-bar'>
-          <Main onErrorButtonClick={() => setHasError(true)}>
+          <Main showNothingFound={isError}>
+            {isFetching && !isLoading && <div style={{color: 'white'}}>Updating...</div>}
             {isLoading ? (
               <Spinner />
-            ) : characters.length === 0 ? null : (
-              <>
-                {characters.map((character, index) => (
-                  <Card
-                    key={index}
-                    character={character}
-                    onClick={() => handleCardClick(character)}
-                  />
-                ))}
-              </>
-            )}
+            ) : data?.results?.length ? (
+              data.results.map((character, i) => (
+                <Card key={i} character={character} onClick={() => handleCardClick(character)} />
```

Comment:
```suggestion
              data.results.map((c) => (
                <Card key={c.id} character={character} onClick={() => handleCardClick(character)} />
```
https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

Reply:
Done